Scripts/Pointer/BasePointer.cs

```diff
@@ -0,0 +1,298 @@
+        /// <summary>
+        /// The duration that the pointer has been hovering over it's current target.
+        /// </summary>
+        public float hoverDuration;
+    }
+
+    /// <summary>
+    /// The BasePointer forms the basis for all pointer types.
+    /// </summary>
+    public abstract class BasePointer : MonoBehaviour
+    {
+        /// <summary>
+        /// The visibility of the pointer element.
+        /// </summary>
+        public enum VisibilityType
+        {
+            /// <summary>
+            /// The element will only be visible when the pointer is in the active state.
+            /// </summary>
+            OnWhenActive,
+            /// <summary>
+            /// The element will always be visible regardless of the pointer state.
+            /// </summary>
+            AlwaysOn,
+            /// <summary>
+            /// The element will never be visible regardless of the pointer state.
+            /// </summary>
+            AlwaysOff
+        }
+
+        [Header("Base Settings")]
+
+        [Tooltip("An optional ExclusionRule to determine valid and invalid targets based on the set rules.")]
+        public ExclusionRule exclusionRule;
+        [Tooltip("An optional custom PhysicsCast object to affect the Cast of the pointer.")]
+        public PhysicsCast physicsCast;
+
+        /// <summary>
+        /// The PointerUnityEvent emits an event with the PointerData payload and the sender object.
+        /// </summary>
+        [Serializable]
+        public class PointerUnityEvent : UnityEvent<PointerData, object>
+        {
+        };
+
+        [Header("Pointer Events")]
+
+        /// <summary>
+        /// The Activated event is emitted when the pointer becomes active.
+        /// </summary>
+        public PointerUnityEvent Activated = new PointerUnityEvent();
+        /// <summary>
+        /// The Deactivated event is emitted when the pointer becomes inactive.
+        /// </summary>
+        public PointerUnityEvent Deactivated = new PointerUnityEvent();
+        /// <summary>
+        /// The Entered event is emitted when the pointer collides with a new target.
+        /// </summary>
+        public PointerUnityEvent Entered = new PointerUnityEvent();
+        /// <summary>
+        /// The Exited event is emitted when the pointer stops colliding with an existing target.
+        /// </summary>
+        public PointerUnityEvent Exited = new PointerUnityEvent();
+        /// <summary>
+        /// The Hovering event is emitted when the pointer changes its hovering position ove an existing target.
+        /// </summary>
+        public PointerUnityEvent Hovering = new PointerUnityEvent();
+        /// <summary>
+        /// 
+        /// </summary>
+        public PointerUnityEvent Selected = new PointerUnityEvent();
+
+        public abstract bool IsVisible();
+
+        /// <summary>
+        /// Reports the active state of the pointer.
+        /// </summary>
+        public bool Active

```

Comment:
Can we perhaps remove this and use `MonoBehaviour`'s `enabled` state instead? I wonder about the confusion it gives having multiple states like these.

Additionally I'd expect the pointer "to turn off" when I disable the component.

Reply:
`Active` isn't the same as enabled.

enabled is the component is on

`Active` in the case of a pointer is the pointer has been activated via the `Activate()` method

Reply:
What does a Pointer do when `enabled == false` but `active == true`?